### ISSUE TITLE
Shell won't open if treemacs is opened.

### ISSUE THREAD

**Issue by hariseldon78** [syl20bnr/spacemacs#13488]
#### Description :octocat:
Shell won't open if treemacs is opened. 

#### Reproduction guide :beetle:
- Start Emacs
- activate treemacs window with `SPC f t`
- open some file window
- activate shell with `SPC '`

*Observed behaviour:* :eyes: :broken_heart:
I get this error in messages:
```
ad-Advice-split-window-internal: Resizing old window failed
```
and the shell doesn't open. If I close the treemacs panel then the shell opens fine.

*Expected behaviour:* :heart: :smile:
Shell!

#### System Info :computer:
- OS: gnu/linux
- Emacs: 26.3
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. acdb59767)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(emacs-lisp git helm helpful import-js
            (javascript :variables javascript-import-tool 'import-js javascript-fmt-tool 'prettier javascript-repl 'nodejs javascript-backend 'tide)
            (lsp :variables lsp-ui-doc-enable nil)
            multiple-cursors org prettier
            (shell :variables shell-default-shell 'vterm shell-default-position 'right shell-default-width 40 terminal-here-terminal-command
                   (list "/usr/bin/terminator"))
            (treemacs :variables treemacs-lock-width t)
            tide
            (typescript :variables typescript-fmt-on-save t typescript-fmt-tool 'prettier typescript-backend 'tide)
            vue)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS GLIB NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS LIBSYSTEMD LCMS2

#### Backtrace :paw_prints:
```
ad-Advice-split-window-internal: Resizing old window failed
```

**Comment by jr0cket** [comment]
I was unable to replicate this.  My shell layer configuration is 
```     
(shell :variables
            shell-default-shell 'eshell
            shell-default-height 30
            shell-default-position 'bottom)
```

#### System Info :computer:
- OS: gnu/linux
- Emacs: 26.3
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 0a7ab7873)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(asciidoc
 (auto-completion :variables auto-completion-enable-help-tooltip t auto-completion-enable-snippets-in-popup t auto-completion-enable-sort-by-usage t)
 (clojure :variables clojure-toplevel-inside-comment-form t cider-overlays-use-font-lock t clojure-enable-linters 'clj-kondo)
 command-log colors csv emacs-lisp emoji
 (git :variables git-magit-status-fullscreen t magit-diff-refine-hunk 'all)
 github graphviz helm html javascript markdown multiple-cursors
 (org :variables org-enable-github-support t org-enable-bootstrap-support t org-enable-reveal-js-support t org-want-todo-bindings t)
 (ranger :variables ranger-show-preview t ranger-show-hidden t ranger-cleanup-eagerly t ranger-cleanup-on-disable t ranger-ignored-extensions
         '("mkv" "flv" "iso" "mp4"))
 (shell :variables shell-default-shell 'eshell shell-default-height 30 shell-default-position 'bottom)
 (spacemacs-layouts :variables spacemacs-layouts-restrict-spc-tab t persp-autokill-buffer-on-remove 'kill-weak)
 spell-checking
 (syntax-checking :variables syntax-checking-use-original-bitmaps t)
 (treemacs :variables treemacs-indentation 1 treemacs-use-filewatch-mode t treemacs-use-follow-mode t)
 (version-control :variables version-control-diff-tool 'diff-hl version-control-global-margin t))
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS GLIB NOTIFY ACL LIBSELINUX GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM THREADS LIBSYSTEMD LCMS2

**Comment by hariseldon78** [comment]
It appears that the bug is strictly bound to the shell being configured in the "right" position, and there must be already 2 vertical split windows present.
So the reproduction guide becomes:
- configure the shell with right position;
- (maybe configure treemacs with lock-width)
- start emacs
- open treemacs (spc f t)
- open 2 buffer windows (spc w 2)
- open shell (spc apostrophe )

that's it.

**Comment by duianto** [comment]
Confirmed.

It seems to be related to `treemacs-lock-width t`.

The `SPC w 2` step isn't needed:
- it's enough to just open Treemacs `SPC f t`
- then the shell `SPC '`

<details><summary>
Windows 1903
</summary>
<pre>
#### System Info :computer:
- OS: windows-nt
- Emacs: 26.3
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 5280ac232)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(autohotkey command-log dap emacs-lisp git helm html imenu-list javascript
            (markdown :variables markdown-live-preview-engine 'vmd markdown-command "vmd")
            multiple-cursors nim
            (org :variables org-agenda-files
                 '("~/org/notes.org"))
            (shell :variables shell-default-shell 'eshell shell-default-height 30 shell-default-position 'right)
            spell-checking
            (syntax-checking :variables syntax-checking-enable-by-default nil)
            (treemacs :variables treemacs-lock-width t treemacs-use-scope-type 'Perspectives)
            version-control)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS THREADS LCMS2

</pre>
</details>

**Comment by emacs18** [comment]
I cannot reproduce on emacs 27 on ubuntu 20.04.

**Comment by duianto** [comment]
>I cannot reproduce on emacs 27 on ubuntu 20.04.

It's still reproducible in Emacs 27.1 for me in Windows 1903.
```elisp
     (shell :variables shell-default-position 'right)
```
with `treemacs-lock-width` enabled either on startup in the layer config:
```elisp
     (treemacs :variables treemacs-lock-width t)
```
or by toggling it on in the Treemacs window with: `tw`

Then `SPC '` shows:
>split-window: Resizing old window failed

When the Treemacs width is unlocked: `tw`
then the expected `*Default-shell-1*` buffer opens to the right in the Emacs frame.

<details><summary>
Windows 10, System Info (Click to expand)
</summary>
<pre>
#### System Info :computer:
- OS: windows-nt
- Emacs: 27.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. cbeafda39)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(autohotkey command-log emacs-lisp git haskell helm
            (javascript :variables javascript-backend 'lsp)
            lsp multiple-cursors org
            (python :variables python-backend 'lsp python-lsp-server 'mspyls)
            (shell :variables shell-default-position 'right)
            version-control
            (treemacs :variables treemacs-lock-width t))
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY W32NOTIFY ACL GNUTLS LIBXML2 HARFBUZZ ZLIB TOOLKIT_SCROLL_BARS MODULES THREADS JSON PDUMPER LCMS2 GMP

</pre>
</details>

**Comment by emacs18** [comment]
I too see the problem now.  The key issues were shell on right (rather than bottom), and locking of treemacs width.

The error message is coming from `split-window-internal` C function called by `split-window` elisp function.

**Comment by hariseldon78** [comment]
> I'm not sure whether this is a bug or not. It appears to me that user set things up so that the width was not large enough for emacs to split it in two. Sure the error message could have been better, but I'm not sure that we can consider this a bug.

No, my configuration is to open the shell at bottom, not right. So there is nothing to split width-wise. It is clearly a bug.

**Comment by hariseldon78** [comment]
I tried again today, and i see the same behavior happens with `neotree` instead of `treemacs`. 
The fact is that if i open the shell first, and then i open the file manager, then it all works correctly. If i open the file manager and then the shell it gives the error. I understand that the shell panel and the file manager panel are width locked, but the other panels are definitely not, and can be resized up to just 8 characters of width.

So i think it is really a bug, because it works in one order and not in the other.

**Comment by emacs18** [comment]
I took a closer look. It could be a bug in `shell-pop` package. If the following one line change is made in that package, then the problem goes away, i.e., `SPC '` works.

```diff
diff --git a/shell-pop.el b/shell-pop.el
index 9e9fdf4..644b6ee 100644
--- a/shell-pop.el
+++ b/shell-pop.el
@@ -360,7 +360,7 @@ The input format is the same as that of `kbd'."
     (cond
      (shell-pop-full-span
       (split-window
-       (frame-root-window) ; window
+       (selected-window) ; window
        (shell-pop--calculate-window-size) ; size
        (shell-pop--translate-position shell-pop-window-position))) ; side
      (t
```

We run into problem when `split-window` is called with `(frame-root-window)` as the first argument, but we have no problem if it was instead `(selected-window)`. I'm not sure what the problem is, but here is my guess. When `(frame-root-window)` is used, then the size argument given needs to be greater than the width of the treemacs window! That is why I think `split-window` fails. I verified this by first observing that when it fails the code involved is this:

```elisp
(split-window (frame-root-window) 29 'right)
```

However if above is changed to bump up the size parameter as shown here

```elisp
(split-window (frame-root-window) 32 'right)
```

then this works.  The numbers above may be specfic to my current setup only.  Others may need to use different numbers.

So the problem could be one of two. First is that the first argument to `split-window` should always be `(selected-window)`. Second is that first argument can be `(frame-root-window)`, but that the window size argument must be bumped up to include the width of the locked treemacs window width.
